databricks/koalas/frame.py

```diff
@@ -2350,6 +2350,88 @@ def fillna(self, value=None, axis=None, inplace=False):
         else:
             return DataFrame(internal)
 
+    def replace(self, to_replace=None, value=None, subset=None, inplace=False,
+                limit=None, regex=False, method='pad'):
+        """
+        Returns a new DataFrame replacing a value with another value.
+
+        Parameters
+        ----------
+        to_replace : int, float, string, or list
+            Value to be replaced. If the value is a dict, then value is ignored and
+            to_replace must be a mapping from column name (string) to replacement value.
+            The value to be replaced must be an int, float, or string.
+        value : int, float, string, or list
+            Value to use to replace holes. The replacement value must be an int, float,
+            or string. If value is a list, value should be of the same length with to_replace.
+        subset : sting, list
+            Optional list of column names to consider. Columns specified in subset that
+            do not have matching data type are ignored. For example, if value is a string,
+            and subset contains a non-string column, then the non-string column is simply ignored.
+        inplace : boolean, default False
+            Fill in place (do not create a new object)
+
+        Returns
+        -------
+        DataFrame
+            Object after replacement.
+
+        Examples
+        --------
+        >>> df = ks.DataFrame({"name": ['Ironman', 'Captain America', 'Thor', 'Hulk'],
+        ...                    "weapon": ['Mark-45', 'Shield', 'Mjolnir', 'Smash']})
+        >>> df
+                      name   weapon
+        0          Ironman  Mark-45
+        1  Captain America   Shield
+        2             Thor  Mjolnir
+        3             Hulk    Smash
+
+        >>> df.replace('Ironman', 'War-Machine')
+                      name   weapon
+        0      War-Machine  Mark-45
+        1  Captain America   Shield
+        2             Thor  Mjolnir
+        3             Hulk    Smash
+
+        >>> df.replace(['Ironman', 'Captain America'], ['Rescue', 'Hawkeye'], inplace=True)
+        >>> df
+              name   weapon
+        0   Rescue  Mark-45
+        1  Hawkeye   Shield
+        2     Thor  Mjolnir
+        3     Hulk    Smash
+
+        >>> df.replace('Mjolnir', 'Stormbuster', subset='weapon')
+              name       weapon
+        0   Rescue      Mark-45
+        1  Hawkeye       Shield
+        2     Thor  Stormbuster
+        3     Hulk        Smash
+        """
+        if method != 'pad':
+            raise NotImplementedError("replace currently works only for method='pad")
+        if limit is not None:
+            raise NotImplementedError("replace currently works only when limit=None")
+        if regex is not False:
+            raise NotImplementedError("replace currently doesn't supports regex")
+
+        if not isinstance(value, (int, float, str, list, dict)):
+            raise TypeError("Unsupported type {}".format(type(value)))
```

Comment:
I think error happened here ... It's `None` by default and `None` is not an instance of `(int, float, str, list, dict)`. So, it looks it raised an exception.

Reply:
```
        if not isinstance(value, (int, float, str, list, dict)):
>           raise TypeError("Unsupported type {}".format(type(value)))
E           TypeError: Unsupported type <class 'NoneType'>
```